gcip/core/rule.py

```diff
@@ -34,10 +63,47 @@ def __init__(
         self._allow_failure = allow_failure
 
     def never(self) -> Rule:
+        """
+        This method returns a copy of this rule with the `when` attribute set to `WhenStatement.NEVER`.
+
+        This method is intended to be used for predefined rules. For instance you have defined an
+        often used rule `on_master` whose if statement checks if the pipeline is executed on branch
+        `master`. Then you can either run a job, if on master...
+
+        ```
+        my_job.append_rules(on_master)
+        ```
+
+        ... or do not run a job if on master...
+
+        ```
+        my_job.append_rules(on_master.never())
+        ```
+
+        Returns:
+            Rule: A new rule object with `when` set to `WhenStatement.NEVER`.
+        """
         self._when = WhenStatement.NEVER
         return self
 
+    def copy(self) -> Rule:
+        """
```

Comment:
I there a need for copy? If I instantiate a new Object of this class, i get a new rule. 
The user is responsible for copying the object if needed.

Reply:
It is used by the `.never()` method. It is the same mechanism used in the `Job` class for example, that is why I haven't made this method private.

But you're right, it is worth thinking about making all `copy()` methods in the gcip private for only internal use: #78

Reply:
Thumbsup? Thumbsdown? Resolve?